db/blob/blob_contents.cc

```diff
@@ -0,0 +1,63 @@
+//  Copyright (c) Meta Platforms, Inc. and affiliates.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "db/blob/blob_contents.h"
+
+#include <cassert>
+
+#include "cache/cache_helpers.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+std::unique_ptr<BlobContents> BlobContents::Create(
+    CacheAllocationPtr&& allocation, size_t size) {
+  return std::unique_ptr<BlobContents>(
```

Comment:
Can we use `std::make_unique`?

Reply:
Unfortunately, `make_unique` / `make_shared` doesn't work when the constructor is private, and making them a `friend` isn't an option either because the name of the actual low-level method that would call the constructor is implementation-defined.